sys/net/application_layer/gcoap/fileserver.c

```diff
@@ -32,14 +32,24 @@
 /** Maximum length of an expressible path, including the trailing 0 character. */
 #define COAPFILESERVER_PATH_MAX (64)
 
+struct requestoptions {
+    uint32_t etag;
+    uint32_t if_match;
+    char if_match_len;
+    struct {
+        bool if_match        :1;
+        bool if_none_match   :1;
+        bool etag            :1;
+        bool block2          :1;
+        bool block1          :1;
+    } exists;
```

Comment:
Some comments would be good here, it's not obvious when those flags would be set or what `if_match` is